### ISSUE TITLE
Performance comparison

### ISSUE THREAD

**Issue by veged** [cdglabs/ohm#34]
It will be great to have some performance comparison between OHM, Ometa/JS and custom parsers (such as Esprima).

**Comment by alexwarth** [comment]
Hi Sergey,

If this is something that you're interested in, then by all means go ahead ;)

Here's my take on performance: it will be very nice to have a super fast implementation of Ohm *eventually*. Right now, however, I'm much more interested in usability: making improvements to the language, its associated tools (e.g., Pat's visualizer), etc. I get that Ohm has to be reasonably efficient in order to be usable / useful, but it doesn't have to be super fast.

My worry is that any additional complexity in the implementation -- which tends to creeps in when people optimize software -- would hinder our ability to make usability improvements and experiment with changes to the language. Having said that, if there are any optimizations that can be made without making the implementation (much) more complicated, great!

Cheers,
Alex

**Comment by bd82** [comment]
Hello.

> It will be great to have some performance comparison between OHM, Ometa/JS and custom parsers (such as Esprima).

I've authored such a benchmark and while it does not include Ometa, it does include 
**eight** different JS parsing libraries including Ohm **and** and a custom
hand built parser.

The Benchmark: http://sap.github.io/chevrotain/performance/

The grammar tested is JSON due to its simplicity and readily available implementations using different parsing libraries.

**Comment by airhorns** [comment]
@alexwarth has Ohm stabilized enough such that you folks would start accepting performance enhancing PRs that might add a little implementation complexity?

**Comment by alexwarth** [comment]
@airhorns: Yes, I think now is a good time. I'd be happy to take a look at any contributions you (or others) would like to make. Thanks! :)

**Comment by ajbouh** [comment]
In the spirit of the above, has anyone identified any low hanging fruit for improving performance?
